connectionprivate.cpp

```diff
@@ -48,43 +48,46 @@ ConnectionPrivate::~ConnectionPrivate()
 
 void ConnectionPrivate::processState(State* state)
 {
-    QString roomId = state->event()->roomId();
     if( state->event()->type() == QMatrixClient::EventType::RoomMember )
     {
         QMatrixClient::RoomMemberEvent* e = static_cast<QMatrixClient::RoomMemberEvent*>(state->event());
         User* user = q->user(e->userId());
         user->processEvent(e);
     }
-    if( !roomId.isEmpty() )
-    {
-        Room* room;
-        if( !roomMap.contains(roomId) )
-        {
-            room = q->createRoom(roomId);
-            roomMap.insert( roomId, room );
-            emit q->newRoom(room);
-        } else {
-            room = roomMap.value(roomId);
-        }
-        room->addInitialState(state);
-    }
+
+    if ( Room* r = provideRoom(state->event()->roomId()) )
+        r->addInitialState(state);
 }
 
 void ConnectionPrivate::processRooms(const QList<SyncRoomData>& data)
 {
     for( const SyncRoomData& roomData: data )
     {
-        Room* room;
-        if( !roomMap.contains(roomData.roomId) )
-        {
-            room = q->createRoom(roomData.roomId);
-            roomMap.insert( roomData.roomId, room );
-            emit q->newRoom(room);
-        } else {
-            room = roomMap.value(roomData.roomId);
-        }
-        room->updateData(roomData);
+        if ( Room* r = provideRoom(roomData.roomId) )
+            r->updateData(roomData);
+    }
+}
+
+Room* ConnectionPrivate::provideRoom(QString id)
+{
+    if (id.isEmpty())
+    {
+        qDebug() << "ConnectionPrivate::provideRoom() with empty id, doing nothing";
+        return nullptr;
     }
+
+    Room* room = roomMap.value(id, nullptr);
```

Comment:
I think it would be easier to read this way:
```
if( roomMap.contains(id) )
    return roomMap.value(id);
```
It might be a bit slower than your version, but I don't think it matters much (it's a hashmap, so we have O(1) complexity)

What do you think?